backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -515,16 +517,26 @@ public boolean unsubscribeEmail(
             )
     })
     @RolesAllowed(ConsoleIdentityProvider.RBAC_WRITE_INTEGRATIONS_ENDPOINTS)
-    public void testEndpoint(@Context SecurityContext sec, @RestPath UUID uuid) {
+    public void testEndpoint(@Context SecurityContext sec, @RestPath UUID uuid, @RequestBody EndpointTestRequest requestBody) {
         final String orgId = SecurityContextUtil.getOrgId(sec);
 
         if (!this.endpointRepository.existsByUuidAndOrgId(uuid, orgId)) {
             throw new NotFoundException("integration not found");
         }
 
-        final var endpointTestRequest = new EndpointTestRequest(uuid, orgId);
+        final InternalEndpointTestRequest internalEndpointTestRequest;
+        if (requestBody == null) {
+            internalEndpointTestRequest = new InternalEndpointTestRequest(uuid, null, orgId);
+        } else {
+            if (requestBody.isMessageBlank()) {
+                throw new BadRequestException("the custom message cannot be empty");
+            }
```

Comment:
If `EndpointTestRequest` is needed, this validation could be done with a `@NotBlank` annotation on `EndpointTestRequest#message` and the `@Valid` annotation on `requestBody`.

Reply:
True! It's way better and cleaner in the way you suggest it. Done!